components/director/internal/domain/application/service.go

```diff
@@ -500,6 +503,35 @@ func (s *service) Upsert(ctx context.Context, in model.ApplicationRegisterInput)
 	return s.genericUpsert(ctx, tenant, in, upserterFunc)
 }
 
+// TryUpdateBaseURL Gets application by ID. If the application does not have a BaseURL set, the API TargetURL is parsed and set as BaseURL
+func (s *service) TryUpdateBaseURL(ctx context.Context, appID, targetURL string) error {
+	appTenant, err := tenant.LoadFromContext(ctx)
+	if err != nil {
+		return errors.Wrapf(err, "while loading tenant from context")
+	}
+
+	app, err := s.Get(ctx, appID)
+	if err != nil {
+		return err
+	}
+
+	if app.BaseURL != nil && len(*app.BaseURL) > 0 {
+		log.C(ctx).Infof("BaseURL for Application %s already exists. Will not update it.", appID)
+		return nil
+	}
+
+	log.C(ctx).Infof("BaseURL for Application %s does not exist. Will update it.", appID)
+
+	parsedTargetURL, err := url.Parse(targetURL)
+	if err != nil {
+		return errors.Wrapf(err, "while parsing targetURL")
+	}
+
+	app.BaseURL = str.Ptr(fmt.Sprintf("%s://%s", parsedTargetURL.Scheme, parsedTargetURL.Host))
```

Comment:
This additional conversion will not be needed if define BaseURL as string, not as pointer in model.Application struct.